### PR TITLE
Migrate Python to batch core API

### LINKED ISSUE #549: Move Python to the new batch API
This is a tracking bug so I know when things are done

### PR DESCRIPTION
**Pull request by soltanmm**
@nathanielmanistaatgoogle Please review and merge!

The `_adapter` layer will change drastically at some point before the next release cycle - this just gets us off of the old API.

Fixes #549

### CODE REVIEW

**Comment by ctiller** [comment]
Thankyou!

File: src/python/src/grpc/_adapter/_tag.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include "grpc/_adapter/_tag.h"
+
+#include <assert.h>
```
Comment (nathanielmanistaatgoogle) [inline src/python/src/grpc/_adapter/_tag.c]: How is this include used in this file?

File: src/python/src/grpc/_adapter/_tag.c [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#include <stdarg.h>
```
Comment (nathanielmanistaatgoogle) [inline src/python/src/grpc/_adapter/_tag.c]: How is this include used in this file?

File: src/python/src/grpc/_adapter/_call.c [hunk 3]
```diff
@@ -33,93 +33,170 @@
 
 #include "grpc/_adapter/_call.h"
 
+#include <stdio.h>
```
Comment (nathanielmanistaatgoogle) [inline src/python/src/grpc/_adapter/_call.c]: How is this import used in this file?

File: src/python/src/grpc/_adapter/_tag.c [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    self->user_tag = user_tag;
+  }
+  Py_INCREF(self->user_tag);
+  self->type = type;
+  self->call = call;
+  Py_INCREF(call);
+  return self;
+}
+
+pygrpc_tag *pygrpc_tag_new_server_rpc_call(PyObject *user_tag) {
```
Comment (nathanielmanistaatgoogle) [inline src/python/src/grpc/_adapter/_tag.c]: Could this function be eliminated by changing its one call site to allocate a Call and pass that Call and PYGRPC_SERVER_RPC_NEW to pygrpc_tag_new? Is this just a convenience function?

**Comment by nathanielmanistaatgoogle** [comment]
Looks mostly good; I'm not sure that the answers to any of my questions will require tweaks to the code.

File: src/python/src/grpc/_adapter/_tag.c (hunk 4 above)
Comment (soltanmm) [inline src/python/src/grpc/_adapter/_tag.c]: Purely a convenience function. Then again, aren't all functions more or less convenience functions? :-P

File: src/python/src/grpc/_adapter/_call.c (hunk 3 above)
Comment (soltanmm) [inline src/python/src/grpc/_adapter/_call.c]: Blargh. Debug cruft.

**Comment by nathanielmanistaatgoogle** [comment]
Looks good; will merge when tests pass.

**Comment by murgatroid99** [comment]
This is a prerequisite for #1397 (Making this comment so that you can see the status of this PR from that one).

**Pull request merged**